### PR TITLE
Highlight features

### PR DESCRIPTION
**Pull request by umututku03**
Hello Professor Liu,

I hope you are doing well! Me and Mimis worked together on two main subjects: Highlight features and blank spaces. We have modified our implementations so that the user can pass in style configurations ( along with default style configurations). For blank spaces, we have modified our implementation for automation algorithms so that the user can specify a blank space on the canvas. Lastly, I have added the documentation (the explanation for the algorithm) of the `drawAutomatedStackFrames`.

See you soon!

Sincerely,
Utku

### CODE REVIEW

File: src/user_functions.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ function draw(objects, automation, configuration) {
 
     if (automation) {
         if (!configuration.hasOwnProperty("width")){
-            throw new Error("Width argument for automated drawing is required.")
+            throw new Error("Width argument for automated.md drawing is required.")
```
Comment (david-yz-liu) [inline src/user_functions.js:43]: this change shouldn't be there

File: examples/demo_7.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    {"isClass": false, "name": "bool", "id": 32, "value": true},
+    {"isClass": false, "name": "str", "id": 43, "value": "David is cool"},
+    {"isClass": false, "name": "tuple", "id": 11, "value": [82, 76]},
+    {"isClass": false, "name": "set", "id": 90, "value": [36, 49, 64]},
+    {"isClass": false, "name": "dict", "id": 10, "value": {"x": 81, "y": 100, "z": 121}},
+    {"isClass": false, "name": "None", "id": 13, "value": "None"}
+
+]
+
+console.log("HERE WE ARE 1!")
```
Comment (david-yz-liu) [inline examples/demo_7.js:42]: Remove all `console.log` calls (in the demo code and in the actual implementation code)

File: src/automate.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 const {MemoryModel, config} = require("./memory_model.js");
 
 /**
- * Draws the objects given in the path in an automated fashion.
+ * Draws the objects given in the path in an automated.md fashion.
```
Comment (david-yz-liu) [inline src/automate.js:4]: I'm not sure what happened, but you seem to have renamed `automated` to `automated.md` in at least two places. Make sure to revert all such changes

File: src/automate.js [hunk 4]
```diff
@@ -42,7 +45,6 @@ function drawAutomated(objects, width, configuration) {
  * @param {Object[]} stack_frames - The list of stack-frames that will be drawn
  * (without the specified x and y coordinates)
  *
- * @returns {Object} -
```
Comment (david-yz-liu) [inline src/automate.js:45]: Don't delete this (if it needs to be updated, then update it)

File: src/automate.js [hunk 5]
```diff
@@ -59,28 +61,46 @@ function drawAutomatedStackFrames(stack_frames, configuration) {
...
     for (const stack_frame of stack_frames){
 
+        // Initialize the width and the height of the stack frames
+        let width;
+        let height;
+
         // get the width and height of each box
-        const {height, width} = getSize(stack_frame)
+        if (stack_frame.name !== 'BLANK') {
+            height = getSize(stack_frame).height;
```
Comment (david-yz-liu) [inline src/automate.js:77]: Don't repeat function calls unnecessarily. `getSize` should only be called once here (as it was before)

File: src/automate.js [hunk 6]
```diff
@@ -127,11 +151,22 @@ function drawAutomatedOtherItems(objs, max_width, sort_by, config_aut = {} /* to
...
     for (const item of objs) {
-        const dimensions = getSize(item);
-        item.height = dimensions.height;
-        item.width = dimensions.width;
+        if (item.name !== "BLANK") {
+            const dimensions = getSize(item);
+            item.height = dimensions.height;
+            item.width = dimensions.width;
+        }
+        else {
```
Comment (david-yz-liu) [inline src/automate.js:159]: Put `else` on the same line as the `}`

File: src/user_functions.js [hunk 7]
```diff
@@ -19,7 +19,7 @@ const fs = require("fs");
  *
  * @returns {MemoryModel} the produced canvas
  */
-function draw(objects, automation, configuration) {
+function draw(objects, automation, configuration, style) {
```
Comment (david-yz-liu) [inline src/user_functions.js:22]: Whenever you add a parameter, the function documentation also needs to be updated

File: src/memory_model.js [hunk 8]
```diff
@@ -156,19 +163,22 @@ class MemoryModel {
             display_text = JSON.stringify(value)
         }
 
+        // config.primitive_default_style
```
Comment (david-yz-liu) [inline src/memory_model.js:166]: Remove commented out code

File: src/memory_model.js [hunk 9]
```diff
@@ -156,19 +163,22 @@ class MemoryModel {
...
+
         // Actually drawing the text to be displayed on our canvas by utilizing the helper 'drawText' instance method.
         // Note that if the value is null or undefined, nothing will be drawn
         if (value !== null && value !== undefined) {
+
```
Comment (david-yz-liu) [inline src/memory_model.js:171]: Don't add additional blank lines

**Review by david-yz-liu — COMMENTED** [review]
@umututku03 @mimischly7 you're on the right track. Please review and clean up your code; I left some comments, but in general there are issues about style and formatting that you should both fix up before I look further. Make sure to remove all commented out code.

File: src/automate.js (hunk 5 above)
Comment (david-yz-liu) [reply src/automate.js:75]: @umututku03 it looks like you resolved this comment (and others) without actually fixing them. That's not good.

Comment (umututku03) [reply src/automate.js:75]: @david-yz-liu Hello Professor Liu, it looks like we did not commit (and push) those changes to the branch that we have created the pull request. I am sorry for the confusion. We definitely intended to actually resolve these comments. Resolving those without actually fixing them is something that is strictly against our work ethic.

**Review by david-yz-liu — APPROVED** [review]

**Pull request merged**